File: frontend/src/shared/types/stock.ts
export interface FactorItem {
  name: string
  weight: number
  raw_value: number
  weighted_contribution: number
  signal: "positive" | "negative"
}

export interface PositionBreakdown {
  base_pct: number
  conviction_multiplier: number
  volatility_adjustment: number
  final_pct: number
}

export interface FactorBreakdown {
  factors: FactorItem[]
  roc: number
  asymmetry_index: number
  time_stop_months: number
  position_breakdown: PositionBreakdown
}

export interface StockScoreResponse {
  stock_symbol: string
  composite_score: number
  signal: string
  position_size: number
  computation_timestamp: string
  kite_snapshot_ts: string
  screener_csv_ts: string
  rbi_csv_ts: string
  factor_breakdown: FactorBreakdown
}

export interface StockListItem {
  stock_symbol: string
  name: string
  signal: string | null
}

export interface StockListResponse {
  items: StockListItem[]
  total: number
}

File: backend/app/schemas/stock.py
from pydantic import BaseModel


class FactorItem(BaseModel):
    name: str
    weight: float
    raw_value: float
    weighted_contribution: float
    signal: str  # "positive" | "negative"


class PositionBreakdown(BaseModel):
    base_pct: float
    conviction_multiplier: float
    volatility_adjustment: float
    final_pct: float


class FactorBreakdown(BaseModel):
    factors: list[FactorItem]
    roc: float
    asymmetry_index: float
    time_stop_months: int
    position_breakdown: PositionBreakdown


class StockScoreResponse(BaseModel):
    stock_symbol: str
    composite_score: float
    signal: str
    position_size: float
    computation_timestamp: str   # ISO 8601 UTC
    kite_snapshot_ts: str        # ISO 8601 UTC
    screener_csv_ts: str         # ISO 8601 UTC
    rbi_csv_ts: str              # ISO 8601 UTC
    factor_breakdown: FactorBreakdown
